src/critic.lisp

```diff
@@ -140,9 +140,23 @@
     (values problems-count)))
 
 
+(defparameter pattern-names
+  (mapcar #'(lambda (rule) (string-downcase (symbol-name rule)))
+          (lisp-critic:get-pattern-names))
+  "List of all LISP-CRITIC pattern names as lowercase strings")
+
+(defun get-blacklist (whitelist)
+  "Returns list of all LISP-CRITIC patterns, excluding WHITELIST arguments."
+  (let ((blacklist nil))
+    (dolist (pattern pattern-names)
+      (unless (member pattern whitelist :test #'string=)
+        (push pattern blacklist)))
+    blacklist))
```

Comment:
Personally, I prefer LOOP for such things as collection. It will be more concise:

```lisp
(loop for pattern in *pattern-names*
          unless (member pattern whitelist :test #'string=)
             collect pattern)
```

And knowing, that STRING= and STRING-EQUAL also works with `string designators` (strings, characters and symbols, let us get rid of global variable `*pattern-names*` and use `(lisp-critic:get-pattern-names)` instead:

```lisp
(loop for pattern in (lisp-critic:get-pattern-names)
          unless (member pattern whitelist :test #'string-equal)
             collect (string-downcase pattern))
```

This will do case insensitive search and string collection in tree lines of code! :)

Reply:
> This will do case insensitive search and string collection in three lines of code! :)

Added your `loop` now in the code and deleted the global parameter. From 8 lines to 3 is quite decent simplification 👍🏻